modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/snapshot/GlobalSnapshotStateChannelManagerSuite.scala

```diff
@@ -0,0 +1,196 @@
+      _ <- manager.accept(SnapshotOrdinal(10L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2))
+      result <- manager.accept(SnapshotOrdinal(11L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2))
+      expected = (SortedMap(address -> NonEmptyList.one(stateChannelOutput2.snapshotBinary)), Set.empty)
+    } yield expect.same(expected, result)
+
+  }
+
+  test("state channel with more sent events should be accepted and other should be discarded") { res =>
+    implicit val (kryo, sp) = res
+    val address = Address("DAG0y4eLqhhXUafeE3mgBstezPTnr8L3tZjAtMWB")
+
+    for {
+      stateChannelOutput1 <- mkStateChannelOutput(1, address, None)
+      stateChannelOutput2 <- mkStateChannelOutput(1, address, None, stateChannelOutput1.snapshotBinary.value.some)
+      stateChannelOutput3 <- mkStateChannelOutput(1, address, None)
+      snapshotInfo = mkGlobalSnapshotInfo(SortedMap.empty)
+      manager <- mkManager()
+      _ <- manager.accept(SnapshotOrdinal(1L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2, stateChannelOutput3))
+      result <- manager.accept(SnapshotOrdinal(11L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2, stateChannelOutput3))
+      expected1 = (SortedMap(address -> NonEmptyList.one(stateChannelOutput2.snapshotBinary)), Set.empty)
+      expected2 = (SortedMap(address -> NonEmptyList.one(stateChannelOutput1.snapshotBinary)), Set.empty)
+    } yield expect.same(expected1, result) || expect.same(expected2, result)
+
+  }
+
+  test("state channels and proper following state channels should be accepted") { res =>
+    implicit val (kryo, sp) = res
+    val address = Address("DAG0y4eLqhhXUafeE3mgBstezPTnr8L3tZjAtMWB")
+
+    for {
+      output1 <- mkStateChannelOutput(1, address, None)
+      output1Hash <- output1.snapshotBinary.toHashed.map(_.hash)
+      output1Following1 <- mkStateChannelOutput(10, address, output1Hash.some)
+      output2 <- mkStateChannelOutput(2, address, None)
+      output2Hash <- output2.snapshotBinary.toHashed.map(_.hash)
+      output2Following1 <- mkStateChannelOutput(5, address, output2Hash.some)
+      output2Following2 <- mkStateChannelOutput(2, address, output2Hash.some)
+      snapshotInfo = mkGlobalSnapshotInfo(SortedMap.empty)
+      manager <- mkManager()
+      _ <- manager.accept(
+        SnapshotOrdinal(1L),
+        snapshotInfo,
+        List(output1, output1Following1, output2, output2Following1, output2Following2)
+      )
+      result <- manager.accept(
+        SnapshotOrdinal(11L),
+        snapshotInfo,
+        List(output1, output1Following1, output2, output2Following1, output2Following2)
+      )
+      expected = (SortedMap(address -> NonEmptyList.of(output2Following1.snapshotBinary, output2.snapshotBinary)), Set.empty)
+    } yield expect.same(expected, result)
+  }
+
+  test("state channels should be accepted despite ordinal when delay not set") { res =>
```

Comment:
`valid state channel genesis snapshot should be always accepted`?

Reply:
Rephrased.